src/lib/API/api.js

```diff
@@ -505,6 +506,78 @@ export class APIService {
 
     return this.sharedClient.post(url, payload, options)
   }
+
+  async tatumQuery(address, chainId, from = null): Promise<any[]> {
+    const url = '/data/transactions'
+    const { CELO, MAINNET, GOERLI } = NETWORK_ID
+
+    let chain
+    const txs = []
+
+    switch (chainId) {
+      case MAINNET:
+        chain = 'ethereum'
+        break
+      case GOERLI:
+        chain = 'ethereum-goerli'
+        break
+      case CELO:
+        chain = 'celo'
+        break
+
+      // FUSE not supported on Tatum
+      default:
+        throw new Error('Chain not supported')
+    }
+
+    const params = { chain, addresses: address, transactionTypes: 'native', offset: 0 }
+    const options = { baseURL: Config.tatumApiUrl, params }
+
+    if (from) {
+      params.blockFrom = from
+    }
+
+    for (;;) {
+      const chunk = await this.sharedClient // eslint-disable-line no-await-in-loop
+        .get(url, options)
+        .then(({ result }) => result.filter(({ transactionSubtype }) => transactionSubtype !== 'zero-transfer'))
+
+      if (!chunk.length) {
+        break
+      }
+
+      params.offset += 50
+      txs.push(...chunk)
+    }
+
+    return txs
+  }
+
+  async fuseExplorerQuery(address, from = null) {
+    const txs = []
+    const url = '/api'
+    const params = { module: 'account', action: 'txlist', address, page: 1 }
+    const options = { baseURL: fuseNetwork.explorerAPI, params }
+
+    if (from) {
+      params.start_block = from
+    }
+
+    for (;;) {
+      const chunk = await this.sharedClient // eslint-disable-line no-await-in-loop
+        .get(url, options)
+        .then(({ result }) => result.filter(({ value }) => value !== '0'))
+
+      if (!chunk.length) {
```

Comment:
same here

Reply:
also seems like the latest celo explorer query is missing.
need to merge with master again?

Reply:
1. will fix
2. will check